source/fswatch.d

```diff
@@ -109,12 +133,17 @@ struct FileWatch
 		/// Implementation using Win32 API or polling for files
 		FileChangeEvent[] getEvents()
 		{
-			if (isDir && (!path.exists || path.isDir))
+			const pathExists = path.exists; // cached so it is not called twice
+			if (isDir && (!pathExists || path.isDir))
 			{
-				if (!path.exists)
+				// ReadDirectoryChangesW does not report changes to the specified directory 
+				// itself, so 'removeself' is checked manually
+				if (!pathExists)
 				{
 					if (pathHandle)
 					{
+						DWORD receivedBytes;
+						OVERLAPPED overlapObj;

```

Comment:
https://docs.microsoft.com/en-us/windows/desktop/api/ioapiset/nf-ioapiset-getoverlappedresult

> lpOverlapped
> A pointer to an OVERLAPPED structure that was specified when the overlapped operation was started.

I think this should stay as private struct member and not be moved inside the function here.